corehq/apps/hqmedia/static/hqmedia/js/hqmediauploaders.js

```diff
@@ -15,7 +16,145 @@ hqDefine("hqmedia/js/hqmediauploaders", function () {
         return HQMediaUploaders;
     };
 
+    const uploader = function (slug, media_type, options) {
+        assertProperties.assertRequired(options, [
+            'uploadURL', 'uploadParams',
+            'queueTemplate', 'errorsTemplate', 'existingFileTemplate',
+        ]);
+
+        var self = {};
+
+        self.queueTemplate = options.queueTemplate;
+        self.errorsTemplate = options.errorsTemplate;
+        self.existingFileTemplate = options.existingFileTemplate;
+        self.uploadParams = options.uploadParams;
```

Comment:
nit: curious why these need to be assigned here vs always referencing them via `options....`? Do are they part of the uploader's public API?

Reply:
They don't need to be assigned here. I've absorbed a habit of assigning `options` attributes to the object itself, because it's what most of our code does, but I don't think there's a strong reason to do so. Are you figuring it would be more concise to just reference them from `options`?

Reply:
Yes, exactly that.

Reply:
Updated in 793e356f3e6 - I did keep `self.uploadParams`, because the value of that changes and I'd prefer to think of `options` as read-only.